src/ray/raylet/worker_pool.cc

```diff
@@ -423,6 +423,14 @@ std::tuple<Process, StartupToken> WorkerPool::StartWorkerProcess(
 
       worker_command_args.push_back("--serialized-runtime-env-context=" +
                                     serialized_runtime_env_context);
+      if (language == Language::JAVA) {
+        // Due to the prepared command for java is
+        // [pythone_executable, setup_worker, "java", "-cp", "path_to_jars"]
+        // therefore we need to remove the ["java", "-cp", "path_to_jars"] because
+        // we should change the class path in setup worker.
+        worker_command_args.erase(worker_command_args.begin() + 2,
+                                  worker_command_args.begin() + 5);
```

Comment:
it's weird that worker pool has to know about this. Can we move this logic to "setup_worker.py"? 
For longer term, I'd like to move all worker command preparation code to "setup_worker.py"

Reply:
DONE